pcdsdevices/tests/test_signal.py

```diff
@@ -60,6 +60,34 @@ def test_avg_signal():
     assert cb.called
 
 
+def test_avg_signal_with_duration():
+    logger.debug("test_avg_signal_with_duration")
+    sig = Signal(name='raw')
+    avg = AvgSignal(sig, 2, name='avg', duration=2)
+
+    assert avg.averages == 2
+
+    sig.put(1)
+    sig.put(3)
+    assert avg.get() == 2
+
+    # trigger should clear the buffer and not take the full duration to return
+    start_time = time.monotonic()
+    st = avg.trigger()
+    end_time = time.monotonic()
+    assert end_time - start_time < avg.duration
+    assert not st.done
```

Comment:
Is there ever a concern that the runner hardware is so bad that the trigger finishes before the puts are done?
2 seconds seems sufficient, but I'm left wondering if there's a better way to ensure it's never a problem.

I can't think of anything better after about a minute of navel gazing

Reply:
I think there will always be some fundamental problem wrt unit tests when coding something with a duration.
Maybe some alternate formulation would wait until the buffer is full instead of a fixed time interval, and then this is much easier to make consistent in the test suite.

I could make the duration longer but then we're hitting tradeoffs with test suite speed.

Reply:
Also to clarify: duration is used in this test to:
- Verify that the trigger returns quickly, instead of waiting
- Verify that the final status after waiting the duration doesn't raise